File: source_funcs/go_transcoder/FIND_A_TRIPLET_THAT_SUM_TO_A_GIVEN_VALUE_1/FIND_A_TRIPLET_THAT_SUM_TO_A_GIVEN_VALUE_1.go
package main

import (
	"fmt"
	"sort"
)

func f_gold(A []int, arr_size int, sum int) bool {
	var (
		l int
		r int
	)
	sort.Ints(A)
	for i := int(0); i < arr_size-2; i++ {
		l = i + 1
		r = arr_size - 1
		for l < r {
			if A[i]+A[l]+A[r] == sum {
				fmt.Print("Triplet is %d, %d, %d", A[i], A[l], A[r])
				return true
			} else if A[i]+A[l]+A[r] < sum {
				l++
			} else {
				r--
			}
		}
	}
	return false
}

File: rust_funcs/c_transcoder/FIND_A_TRIPLET_THAT_SUM_TO_A_GIVEN_VALUE_1/FIND_A_TRIPLET_THAT_SUM_TO_A_GIVEN_VALUE_1.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


fn f_gold(A: &mut [f32], arr_size: i32, sum: i32) -> i32 {
    let mut i = 0;
    let mut l = 1;
    let mut r = arr_size - 1;
    A.sort();
    while i < arr_size - 2 {
        if A[i] + A[l] + A[r] == sum as f32 {
            println!("Triplet is {}, {}, {}", A[i], A[l], A[r]);
            return 1;
        } else if A[i] + A[l] + A[r] < sum as f32 {
            l += 1;
        } else {
            r -= 1;
        }
        i += 1;
    }
    0
}
